### DISCUSSION TITLE
[Bug]: clang: error: linker command failed with exit code 1 (use -v to see invocation)

### DISCUSSION THREAD

**Discussion by yangqingyuan** [alibaba/xquic#39] [Q&A] [answered]
### What happened?

编译过程中出现如下错误，使用babassl跟boringssl的方式都会有相同的错误

电脑配置如下：
mac 11.5.2 (20G95)
Mac mini (2018)
3 GHz 六核Intel Core i5

错误信息如下：
ld: unknown option: --whole-archive
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libxquic.dylib] Error 1
make[1]: *** [CMakeFiles/xquic.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 90%] Built target xquic-static

详细如下图
![screenshot-20220112-102810](https://user-images.githubusercontent.com/6867757/149053034-81d14141-bb56-46fb-b0e2-bb60eed984f7.png)
.

### Steps To Reproduce

按照QuickStart Guide的脚本进行编译，

注意：脚本中“cd ./third_party/babassl/” 应该是“cd ../third_party/babassl/”，不是一个点而是两个点

**Comment by ruiqizhou** [comment, marked as answer]
You can use the cmake variables `-DPLATFORM=mac` to build xquic on MacOS.
https://github.com/alibaba/xquic/blob/main/docs/Platforms.md#macos-release
